### PR TITLE
PHN7 - Add A Twist Mux

### LINKED ISSUE #7: Add Twist Mux
**Description:**
- Sometimes it is needed to change which node is publishing to `cmd_vel`, which is where a multiplexer comes in handy

**Goals:**
- [ ] Multiplex b/w different `cmd_vel` topics
  -  Keyboard teleop (top priority)
  - `rqt_steering`
  - Navigation stack (low priority)

**Acceptance Criteria:**
- [ ] Can control Photon with keyboard even when `rqt_steering` is moving the robot or during an autonomous navigation session

**Notes:**
- Here is a reference for the twist multiplexer: http://wiki.ros.org/twist_mux

### PR DESCRIPTION
**Pull request by cuevasm2**
### Description ###

Added a twist-mux in the folder twist/twist_mux. There are two topics for input to the twist mux: `nav_vel` (for autonomous navigation) and `man_vel` (for manual control). The output `cmd_vel` has been updated to re-route to the one used by the ackermann steering controller.

Files Edited:
~twist/twist_mux/twist_mux_topics.yaml~ `twist/twist_mux/photon_twist_mux_topics.yaml` (*NEW*)
~twist/twist_mux/twist_mux_locks.yaml~ `twist/twist_mux/photon_twist_mux_locks.yaml` (*NEW*)
`description/simulate.launch`

~Note the following files are from the original git:
twist/twist_mux/twist_mux_topics_copy.yaml
twist/twist_mux/twist_mux_locks_copy.yaml~
Original files are left intact!

To test:
In one terminal, this will create twist-mux with rqt steering:
```roslaunch description simulate.launch```
In second terminal,
```rqt```
In new window, go to Plugins->Robot Tools->Robot Steering
enter the key_vel topic in the text box. 

### Issues ###

Priorities between topics cmd_vel and key_vel are not clear.

### Notes ###

Robot is still shaking when steering.

### CODE REVIEW

File: description/launch/spawn.launch [hunk 1]
```diff
@@ -28,6 +28,11 @@
         <node name="joint_state_publisher" pkg="joint_state_publisher" type="joint_state_publisher"/>
         <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher"/>
         
+        <!-- Added rqt robot steering -->
+        <node name="rqt_robot_steering" pkg="rqt_robot_steering" type="rqt_robot_steering">
+        <param name="default_topic" value="/cmd_vel"/>
```
Comment (kishya) [inline description/launch/spawn.launch]: Change /cmd_vel, to /key_vel since key_vel is higher topic in rqt_graph

**Review by kishya — CHANGES_REQUESTED** [review]

File: description/launch/simulate.launch [hunk 2]
```diff
@@ -8,8 +8,12 @@
 
     <!-- Spawn the robot -->
     <include file="$(find description)/launch/spawn.launch">
-        <arg name="robot_ns" value="photon"/>
+        <!-- <arg name="robot_ns" value="photon"/> -->
     </include>
```
Comment (erickmu1) [inline description/launch/simulate.launch]: remove since not using namespace

File: description/launch/simulate.launch [hunk 3]
```diff
@@ -8,8 +8,12 @@
...
 
+    <include file="$(find twist_mux)/launch/twist_mux_custom.launch">
```
Comment (erickmu1) [inline description/launch/simulate.launch]: commen saying what this code launches

File: twist/twist_mux/launch/twist_mux_custom.launch [hunk 4]
```diff
@@ -0,0 +1,25 @@
+<launch>
+  <arg name="joy_vel_in"  default="input_joy/cmd_vel"/>
+  <arg name="joy_vel_out" default="joy_vel"/>
+
+  <arg name="cmd_vel_out" default="twist_mux/cmd_vel"/>
+
+  <arg name="config_locks"  default="$(find twist_mux)/config/twist_mux_locks.yaml"/>
+  <arg name="config_topics" default="$(find twist_mux)/config/twist_mux_topics.yaml"/>
```
Comment (erickmu1) [inline twist/twist_mux/launch/twist_mux_custom.launch]: change name to make clear it for photon + restore original yaml

File: twist/twist_mux/config/twist_mux_topics.yaml [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+
+topics:
+-
+  name    : navigation
+  topic   : nav_vel
+  timeout : 0.5
+  priority: 50
+-
+  name    : keyboard
+  topic   : key_vel
```
Comment (erickmu1) [inline twist/twist_mux/config/twist_mux_topics.yaml:21]: change to `man_vel` for `manual`

**Review by erickmu1 — CHANGES_REQUESTED** [review]

**Review by EsteBran — APPROVED** [review]

File: description/launch/spawn.launch (hunk 1 above)
Comment (erickmu1) [reply description/launch/spawn.launch]: we are not going to use `key_vel`. Rather, using `rqt_robot_steering` is basically manual control, so we want it to publish to `man_vel`.

**Review by erickmu1 — APPROVED** [review]
Looks good! Most new files are just a result of copying the `twist_mux` repo in this branch

**Pull request merged**